src/mpi/errhan/errutil.c

```diff
@@ -218,7 +218,11 @@ int MPIR_Errutil_is_initialized(void)
 /* Return true if the error code indicates a fatal error */
 int MPIR_Err_is_fatal(int errcode)
 {
-    return (errcode & ERROR_FATAL_MASK) ? TRUE : FALSE;
+    if (errcode & ERROR_DYN_MASK) {
+        return FALSE;
+    } else {
+        return (errcode & ERROR_FATAL_MASK) ? TRUE : FALSE;
+    }
```

Comment:
Since `ERROR_DYN_MASK` and `ERROR_FATAL_MASK` are defined to be the same, I don't really understand this change. All fatal codes are going to return FALSE now?